examples/ecs/immutable_components_dynamic.rs

```diff
@@ -0,0 +1,75 @@
+#![allow(unsafe_code)]
+
+//! This example show how you can create immutable components dynamically.
+
+use core::alloc::Layout;
+
+use bevy::{
+    ecs::component::{ComponentDescriptor, ComponentId, StorageType},
+    prelude::*,
+    ptr::OwningPtr,
+};
+
+fn setup(world: &mut World) {
+    // This is a list of dynamic components we will create.
+    // The first item is the name of the component, and the second is the size
+    // in bytes.
+    let my_dynamic_components = [("Foo", 1), ("Bar", 2), ("Baz", 4)];
+
+    // This pipeline takes our component descriptions, registers them, and gets
+    // their ComponentId's.
+    let my_registered_components = my_dynamic_components
+        .into_iter()
+        .map(|(name, size)| {
+            // SAFETY:
+            // - No drop command is required
+            // - The component will store [u8; size], which is Send + Sync
+            let descriptor = unsafe {
+                ComponentDescriptor::new_immutable_with_layout(
```

Comment:
This is all that really changes when working with _immutable_ dynamic components; the `ComponentDescriptor` just needs to include the immutable flag and then everything else works as it did previously.